app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/map/offline/ManageOfflineMapActivityTest.kt

```diff
@@ -0,0 +1,188 @@
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.platform.app.InstrumentationRegistry
+import ch.epfl.sdp.drone3d.R
+import ch.epfl.sdp.drone3d.map.MapboxUtility
+import ch.epfl.sdp.drone3d.map.offline.OfflineMapSaverImplTest
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.offline.OfflineManager
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import org.hamcrest.CoreMatchers.not
+import org.hamcrest.Matcher
+import org.junit.*
+import org.junit.rules.RuleChain
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+
+@HiltAndroidTest
+class ManageOfflineMapActivityTest {
+
+    companion object {
+        private const val TIMEOUT = 5L
+        private const val ZOOM_VALUE = 19.9
+        private val ZOOM_LOCATION = LatLng(10.0, 10.0)
+    }
+
+    private val activityRule = ActivityScenarioRule(ManageOfflineMapActivity::class.java)
+
+    @get:Rule
+    var testRule: RuleChain = RuleChain.outerRule(HiltAndroidRule(this))
+        .around(activityRule)
+
+
+    /**
+     * This object is used to click on a specific element with a given id inside a recyclerView.
+     * It is taken from https://stackoverflow.com/questions/28476507/using-espresso-to-click-view-inside-recyclerview-item
+     */
+    object MyViewAction {
+        fun clickChildViewWithId(id: Int): ViewAction {
+            return object : ViewAction {
+                override fun getConstraints(): Matcher<View>? {
+                    return null
+                }
+
+                override fun getDescription(): String {
+                    return "Click on a child view with specified id."
+                }
+
+                override fun perform(uiController: UiController?, view: View) {
+                    val v: View = view.findViewById(id)
+                    v.performClick()
+                }
+            }
+        }
+    }
+
+    @Before
+    fun setUp() {
+        Intents.init()
+    }
+
+    /**
+     * This method clears the database that stores the OfflineRegions in Mapbox before every test.
+     */
+    @Before
+    fun clearOfflineMapDatabaseBefore(){
+        val counter = CountDownLatch(1)
+        val offlineManager = OfflineManager.getInstance(InstrumentationRegistry.getInstrumentation().targetContext)
+        offlineManager.resetDatabase(object : OfflineManager.FileSourceCallback {
+            override fun onSuccess() {
+                counter.countDown()
+            }
+
+            override fun onError(message: String) {
+
+
+            }
+        })
+
+        assert(counter.await(TIMEOUT, TimeUnit.SECONDS))
+    }
```

Comment:
Why not mock an empty database instead ?

Reply:
As I explained a bit later in the code, we cannot mock the whole OfflineMapSaver mainly because we cannot create OfflineRegion since it has no public constructor. Therefore, we decided to use our implementation (OfflineMapSaverImp) to test the activity and rather do a sort of "end-to-end testing" for this activity. This class uses an OfflineManager which saves the OfflineRegion we want to download in its own Database. To be sure that we start each test with the database with the same state, which makes our testing easier (without any data from an other run, like if you use the app on an emulator first), we clear it.